### PR TITLE
Add chunked media upload.

### PR DESCRIPTION
**Pull request by apizzini**
Added a new method to handle chunked media upload, that's useful for videos or big images (see: https://dev.twitter.com/rest/reference/post/media/upload-chunked).

### CODE REVIEW

**Comment by abraham** [comment]
I'm wondering if maybe it would be better to modify the existing `upload` method?

    public function upload($path, array $parameters = array(), boolean chunked = false)

**Comment by apizzini** [comment]
Yeah, you are right. I'll update the code tomorrow!

**Comment by apizzini** [comment]
I've refactored the code!

File: src/TwitterOAuth.php [hunk 1]
```diff
@@ -225,12 +226,28 @@ public function put($path, array $parameters = array())
...
-    public function upload($path, array $parameters = array())
+    public function upload($path, array $parameters = array(), $chunked = false)
     {
-        $file = file_get_contents($parameters['media']);
-        $base = base64_encode($file);
-        $parameters['media'] = $base;
-        return $this->http('POST', self::UPLOAD_HOST, $path, $parameters);
+        if ($chunked) {
+            // Init
+            $init = $this->http('POST', self::UPLOAD_HOST, $path, ['command' => 'INIT', 'media_type' => $parameters['media_type'], 'total_bytes' => filesize($parameters['media'])]);
```
Comment (0x46616c6b) [inline src/TwitterOAuth.php]: A line should never be longer than 120 characters. It is not so easy to review.

Comment (abraham) [inline src/TwitterOAuth.php]: :+1:

File: src/TwitterOAuth.php [hunk 2]
```diff
@@ -225,12 +226,28 @@ public function put($path, array $parameters = array())
...
      * @return array|object
      */
-    public function upload($path, array $parameters = array())
+    public function upload($path, array $parameters = array(), $chunked = false)
     {
-        $file = file_get_contents($parameters['media']);
-        $base = base64_encode($file);
-        $parameters['media'] = $base;
-        return $this->http('POST', self::UPLOAD_HOST, $path, $parameters);
+        if ($chunked) {
```
Comment (abraham) [inline src/TwitterOAuth.php]: If you've got time I'd like to see the code in the `if` moved into two private functions.

    if ($chunked) {
        // call chunked upload method
    } else {
        // call none chunked upload method
    }

Comment (apizzini) [inline src/TwitterOAuth.php]: Done...

File: src/TwitterOAuth.php (hunk 1 above)
Comment (apizzini) [inline src/TwitterOAuth.php]: Done...

**Comment by abraham** [comment]
Lol, apparently `const UPLOAD_CHUNK = 1024 * 40;` only works in PHP >= 5.6 so I had to update it to `const UPLOAD_CHUNK = 40960;`. :-/

**Pull request merged**